src/Documentation/SidebarMenu/helper.js

```diff
@@ -0,0 +1,149 @@
+import startCase from 'lodash.startcase'
+
+import sidebar from '../sidebar'
+
+/*
+  We will use this helper to normalize sidebar structure and create
+  all of the resurces we need to prevent future recalculations.
+
+  Target structure example:
+
+  {
+    label: "Add Files or Directories",
+    path: "/doc/get-started/add-files",
+    source: "/static/docs/get-started/add-files.md",
+    prev: "/doc/get-started/configure",
+    next: "/doc/get-started/share-data",
+    children: []
+  }
+*/
+
+const PATH_ROOT = '/doc/'
+const FILE_ROOT = '/static/docs/'
+const FILE_EXTENSION = '.md'
+
+// Inner helpers
+
+function findItem(data, targetPath) {
+  if (data.length) {
+    for (let i = 0; i < data.length; i++) {
+      const { path, children } = data[i]
+
+      if (path === targetPath) {
+        return data[i]
+      } else if (children) {
+        const result = findItem(children, targetPath)
+        if (result) {
+          return result
+        }
+      }
+    }
+  }
+}
+
+function findChildWithSource(item) {
+  return item.source ? item : findChildWithSource(item.children[0])
+}
+
+function findPrevItemWithSource(data, item) {
+  if (item.source) {
+    return item
+  } else if (item.prev) {
+    const prevItem = findItem(data, item.prev)
+
+    return findPrevItemWithSource(data, prevItem)
+  }
+}
+
+function validateRawItem({ slug, source, children }) {
+  const isSourceDisabled = source === false // is source set to 'false'?
+
+  if (!slug) {
+    throw Error("'slug' field is required in objects in sidebar.json")
+  }
+
+  if (isSourceDisabled && (!children || !children.length)) {
+    throw Error(
+      "If you set 'source' to false, you had to add at least one child"
+    )
+  }
+}
+
+// Global cache vars used in normalization
+
+let prevReference // Save last item here to generate the prev field
+const normalizedSidebar = [] // Current state of sidebar to search for prev
+
+// Normalization
+
+function normalizeItem(item, parentPath) {
+  validateRawItem(item)
+
+  const { label, slug, source } = item
+
+  // If prev item don't have source we need to recirsively search for it
+  const prevItemWithSource =
+    prevReference && findPrevItemWithSource(normalizedSidebar, prevReference)
+
+  const prev = prevItemWithSource && prevItemWithSource.path
+
+  const sourceFileName = source ? source : slug + FILE_EXTENSION
+  const sourcePath = FILE_ROOT + parentPath + sourceFileName
+
+  return {
+    path: PATH_ROOT + parentPath + slug,
+    source: source === false ? false : sourcePath,
+    label: label ? label : startCase(slug),
+    prev,
+    next: undefined
+  }
+}
+
+function normalizeSidebar(data, parentPath, currentSidebar) {
+  data.forEach(item => {
```

Comment:
ie11? not that it's very important to keep supporting it, but if it's the only that breaks it I would do some workaround

Reply:
forEach works in IE starting with IE9 according to https://caniuse.com/#feat=es5 , so it should not be a problem.